packages/cli-kit/src/public/node/themes/api.test.ts

```diff
@@ -273,3 +275,45 @@ describe('request errors', () => {
     })
   })
 })
+
+describe('bulkUploadThemeAssets', async () => {
+  test('uploads multiple assets', async () => {
+    const id = 123
+    const assets: AssetParams[] = [
+      {key: 'snippets/product-variant-picker.liquid', value: 'content'},
+      {key: 'templates/404.json', value: 'content'},
+    ]
+
+    vi.mocked(restRequest).mockResolvedValue({
+      json: {assets},
+      status: 200,
+      headers: {},
+    })
```

Comment:
Actually, the response we get from the Assets API is slightly different. The HTTP request may return the regular errors handled by [`request`](https://github.com/Shopify/cli/blob/1d9f6de15160a0c2395f6133af1ea8f6c6d6bbe0/packages/cli-kit/src/public/node/themes/api.ts#L74), but the expected state is a `207` code. Also, the body of the response looks like this:

```json
{
  "results": [
    {
      "code": 200,
      "body": {
        "asset": {
          "key": "snippets/header-search.liquid",
          "public_url": null,
          "created_at": "2024-01-25T15:04:01+01:00",
          "updated_at": "2024-01-25T15:04:01+01:00",
          "content_type": "application/x-liquid",
          "size": 4278,
          "checksum": "0603eff236067229c7bc9f9d7de5dd40",
          "theme_id": 162809053206,
          "warnings": []
        }
      }
    },
    {
      "code": 200,
      "body": {
        "asset": {
          "key": "snippets/icon-3d-model.liquid",
          "public_url": null,
          "created_at": "2024-01-25T15:04:02+01:00",
          "updated_at": "2024-01-25T15:04:02+01:00",
          "content_type": "application/x-liquid",
          "size": 1357,
          "checksum": "c6dfd2fd1b62d459121b003ce26173f6",
          "theme_id": 162809053206,
          "warnings": []
        }
      }
    }
  ]
}
```

I believe it's a good idea to return this to function users, so if an error happens for one of the assets, the function users can retry the upload in a new call. We probably want a type representing each item, possibly something like:

```jsonc
{
  key: 'snippets/header-search.liquid',
  success: true,
  errors: [ ... ],
  asset: { key: 'snippets/header-search.liquid', value: '<new_value>', size, ... }
}
```

What do you think about that? :)

Reply:
No problem! Thank **you** for the updates in the PR 🚀 

Why do we need the `size` in the `BulkUploadResult`?

I personally see `BulkUploadResult` and `ThemeAsset` as different entities, so having `size` in the `BulkUploadResult` sounds a bit counterintuitive. Am I missing something though?

Thanks! :)